### DISCUSSION TITLE
Why the name "OpenBao"

### DISCUSSION THREAD

**Discussion by qjoly** [openbao/openbao#34] [Q&A] [answered]
In the case of OpenTofu, the name is easy enough to understand.

But I couldn't find the meaning of OpenBao. Is there one?

**Comment by joewxboy** [comment, marked as answer]
@QJoly BaoZi is a steamed bun, usually filled with meat or vegetables, or a bean paste.  Without being told by the person who made it, there's no way to determine the contents by visually inspecting the bun itself, other than biting into it.  So it's a loose metaphor for an opaque storage system for valuable contents that can reveal itself to the bun's "owner". :). It also has the fun side-effect of continuing the pattern begun by OpenTofu and utilizing asian street food names.  So, Open == open-source software and Bao == BaoZi.

![OpenBao](https://github.com/openbao/openbao/assets/782091/308ca493-e945-4e8d-9508-2205f109c3da)
